### ISSUE TITLE
 .NET Core Hosting Bundle Installer sucks!

### ISSUE THREAD

**Issue by Joebeazelman** [aspnet/IISIntegration#1282]
I spent an entire day diagnosing error 502.5 when I accessed my application. The URL link on the error page is useless.   I checked all the usual suspects including chasing down error logs.  The event viewer had just one, but it didn't provide any clues. Enabling the webapp log files yielded empty log files.  Kestrel, however, was able to serve and respond to requests.  IIS, however, was completely off script.

Initially, I read this useless page on what to look for:

[Troubleshoot ASP.NET Core on IIS](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/troubleshoot?view=aspnetcore-2.1)

It was only after reading painful tomes of redundant and profoundly verbose materials on .NETCore did I finally discover the source of the problem. I had to restart IIS and the WAS service as mentioned in this document:

[Common errors reference for Azure App Service and IIS with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/azure-iis-errors-reference?view=aspnetcore-2.1)

WTF?  Why wasn't this the solution mentioned on the Troubleshooting page?  I lost most of the day chasing a dead end by trying to figure out the relationship between Kestrel (another web server!?) and IIS in order to repair the "reverse proxy configuration" thanks to this:

>  If the app responds normally at the Kestrel endpoint address, the problem is more likely related to the reverse proxy configuration and less likely within the app.

This major time waster could have been solved if the  .NET Core Hosting Bundle Installer offered to restart the server or at the very least alert the user that restarting the server is required.  Ironically, Microsoft installers are known for getting you to restart in the most inconvenient of times, but in this case not restarting allows the Microsoft to punish the user even more.

What were the developers thinking here?  This monumental oversight either proves whiteboard interviews are completely useless or legalization in Seattle has some serious drawbacks.

**Comment by pakrym** [comment]
https://github.com/aspnet/IISIntegration/issues/1278

**Comment by pakrym** [comment]
Closing in favour of the aforementioned issue.
